cmd/kube-proxy/app/server_others.go

```diff
@@ -181,27 +173,48 @@ func newProxyServer(
 
 	klog.V(2).Info("DetectLocalMode: '", string(detectLocalMode), "'")
 
+	primaryProtocol := utiliptables.ProtocolIPv4
+	if utilsnet.IsIPv6(nodeIP) {
+		primaryProtocol = utiliptables.ProtocolIPv6
+	}
+	iptInterface = utiliptables.New(execer, primaryProtocol)
+
+	var ipt [2]utiliptables.Interface
+	dualStack := utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack) && proxyMode != proxyModeUserspace
+	if dualStack {
+		// Create iptables handlers for both families, one is already created
+		// Always ordered as IPv4, IPv6
+		if primaryProtocol == utiliptables.ProtocolIPv4 {
+			ipt[0] = iptInterface
+			ipt[1] = utiliptables.New(execer, utiliptables.ProtocolIPv6)
+
+			// Just because the feature gate is enabled doesn't mean the node
+			// actually supports dual-stack
+			if _, err := ipt[1].ChainExists(utiliptables.TableFilter, utiliptables.ChainInput); err != nil {
```

Comment:
the error in the CI only happens in the table NAT

> > I0212 20:00:52.552777       1 iptables.go:351] running ip6tables-save [-t filter]
> I0212 20:00:52.555445       1 iptables.go:351] running ip6tables-save [-t nat]
> E0212 20:00:52.559906       1 proxier.go:1949] Failed to execute iptables-save, syncing all rules: exit status 1
> E0212 20:00:52.563206       1 proxier.go:1907] Failed to ensure that nat chain KUBE-MARK-DROP exists: error creating chain "KUBE-MARK-DROP": exit status 3: ip6tables v1.8.5 (legacy): can't initialize ip6tables table `nat': Table does not exist (do you need to insmod?)
> Perhaps ip6tables or your kernel needs to be upgraded.
> I0212 20:00:52.625278       1 iptables.go:416] running ip6tables-restore [-w 5 -W 100000 --noflush --counters]
> E0212 20:00:52.629072       1 proxier.go:1619] Failed to execute iptables-restore: exit status 2 (ip6tables-restore v1.8.5 (legacy): ip6tables-restore: unable to initialize table 'nat'
> 
> Error occurred at line: 1
> Try `ip6tables-restore -h' or 'ip6tables-restore --help' for more information.
> )
> Rules:
> *nat

Reply:
that I think for kube-proxy is the more important table to monitor

Reply:
we need to check the nat table, the kube-proxy logs keep showing the same error with this PR

Reply:
Yes, but I assume it fails on `nat` because trying to reference the `nat` table is the very first thing we do ("`Error occurred at line: 1`") not because the problem is specific to `nat`.

Reply:
That said, this PR does not fix the problem.

    I0216 19:46:49.653776       1 server_others.go:137] Detected node IP 10.40.0.5
    I0216 19:46:49.684755       1 server_others.go:203] kube-proxy running in dual-stack mode, %!s(func() iptables.Protocol=0x1786e80)-primary
    ...
    E0216 19:46:50.139766       1 proxier.go:1619] Failed to execute iptables-restore: exit status 2 (ip6tables-restore v1.8.5 (legacy): ip6tables-restore: unable to initialize table 'nat'

but

    Feb 16 19:44:18.154608 localhost kernel: Linux version 5.4.49+ ...

so it's not "the kernel is older than 3.7"

Reply:
the problem is that those nodes doesn't have the the kernel ipv6 nat module loaded,

https://github.com/kubernetes/test-infra/blob/2abb7ff26579325f6a335990003f665755c96d7a/images/krte/wrapper.sh#L77-L78

we had to loaded it in order ip6tables to work

```
  # enable ipv6 iptables
  modprobe -v ip6table_nat
```